AppController/djinn.rb

```diff
@@ -3543,6 +3614,32 @@ def assign_datastore_processes
     }
   end
 
+  def assign_search2_processes
+    # Shadow is the only node to call this method,
+    # and is called upon startup.
+    return unless my_node.is_shadow?
+
+    Djinn.log_info("Assigning search processes.")
+    verbose = @options['verbose'].downcase == 'true'
+
+    # Assign the proper number of Search2 processes on each search2 machine.
+    get_search2.each { |node|
+      assignments = {}
+      begin
+        cpu_count = HermesClient.get_cpu_count(node.private_ip, @@secret)
+        server_count = (cpu_count * Search2::MULTIPLIER).to_i
+        server_count = 1 if server_count == 0
+      rescue FailedNodeException
+        server_count = Search2::DEFAULT_NUM_SERVERS
+      end
+
+      assignments['search'] = {'count' => server_count,
+                               'verbose' => verbose}
+      ZKInterface.set_machine_assignments(node.private_ip, assignments)
```

Comment:
I would recommend adding a version argument to `set_machine_assignments` to ensure the set only takes place if the current version matches the version that was read and modified.

Here's an example since the zookeeper-ruby docs are sparse:
```
node_details = @@zk.get(path: machine_node)
@@zk.set(path: machine_node, data: updated_data, version=node_details[:version])
```